Pipe: Never clear the exceptions for user information & Only record the last exception & Add suggested actions for exceptions

Caideyipi opened the pull request:
## Description
As the title said.

## Test
<img width="3654" height="650" alt="image" src="https://github.com/user-attachments/assets/3ba5be6b-926c-47e3-b62b-161469ec1946" />

<!--
In each section, please describe design decisions made, including:
 - Choice of algorithms
 - Behavioral aspects. What configuration values are acceptable? How are corner cases and error 
    conditions handled, such as when there are insufficient resources?
 - Class organization and design (how the logic is split between classes, inheritance, composition, 
    design patterns)
 - Method organization and design (how the logic is split between methods, parameters and return types)
 - Naming (class, method, API, configuration, HTTP endpoint, names of emitted metrics)
-->


<!-- It's good to describe an alternative design (or mention an alternative name) for every design 
(or naming) decision point and compare the alternatives with the designs that you've implemented 
(or the names you've chosen) to highlight the advantages of the chosen designs and names. -->

<!-- If there was a discussion of the design of the feature implemented in this PR elsewhere 
(e. g. a "Proposal" issue, any other issue, or a thread in the development mailing list), 
link to that discussion from this PR description and explain what have changed in your final design 
compared to your original proposal or the consensus version in the end of the discussion. 
If something hasn't changed since the original discussion, you can omit a detailed discussion of 
those aspects of the design here, perhaps apart from brief mentioning for the sake of readability 
of this PR description. -->

<!-- Some of the aspects mentioned above may be omitted for simple and small changes. -->

<hr>

This PR has:
- [ ] been self-reviewed.
    - [ ] concurrent read
    - [ ] concurrent write
    - [ ] concurrent read and write 
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. 
- [ ] added or updated version, __license__, or notice information
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious 
  for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold 
  for code coverage.
- [ ] added integration tests.
- [ ] been tested in a test IoTDB cluster.

<!-- Check the items by putting "x" in the brackets for the done things. Not all of these items 
apply to every PR. Remove the items which are not done or not relevant to the PR. None of the items 
from the checklist above are strictly necessary, but it would be very helpful if you at least 
self-review the PR. -->

<hr>

##### Key changed/added classes (or packages if there are too many classes) in this PR

Caideyipi committed:
never-clear

Caideyipi committed:
revert

Caideyipi committed:
fix

Caideyipi committed:
shit

Caideyipi committed:
fix

Caideyipi committed:
revert

Caideyipi committed:
fix

Caideyipi committed:
test

Caideyipi committed:
revert

Caideyipi committed:
Merge branch 'master' of https://github.com/apache/iotdb into fix-e

Caideyipi committed:
Merge branch 'master' of https://github.com/apache/iotdb into fix-e

Caideyipi committed:
ger

Caideyipi committed:
fi

Caideyipi committed:
fix

Caideyipi committed:
fix

Caideyipi committed:
try-fix

Caideyipi committed:
Update IoTDBPipePermissionIT.java

Caideyipi committed:
fix

luoluoyuyu reviewed (commented):
## Review 总结

「只保留最后一条异常 + 用户可见时不自动清理 + 建议动作」方向正确。

**合入前请处理**：
1. 修 CI `dual-table-manual-basic` 失败
2. 确认 Thrift/客户端对 `last_exception` 列名变更的兼容性
3. 见行内 2 条实现细节

luoluoyuyu reviewed (commented):
## Review summary

Keeps only the latest pipe exception for display, avoids clearing it automatically, and adds suggested actions. trackException clears then adds a single entry; coordinator meta sync clears exceptions on the agent to avoid stale overwrites.

Please fix failing CI before merge. See inline notes.

luoluoyuyu commented on iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/agent/task/meta/PipeTaskMeta.java:
```diff
@@ -80,42 +81,43 @@ public void setLeaderNodeId(final int leaderNodeId) {
     // Here we still keep the map form to allow compatibility with legacy versions
-    exceptionMessages.clear();
-    exceptionMessages.add(exceptionMessage);
+    lastException.clear();
```
clear then add enforces a single last exception. If SHOW PIPES now exposes last_exception as a structured column, consider whether getExceptionMessage() (Set toString) should be updated or deprecated for user-facing output.

Caideyipi replied:
Kept the existing exception serialization and restored the user-facing column names to ExceptionMessage / exception_message for compatibility. TPipeInfo.exceptionMessage and existing SHOW PIPES / information_schema clients therefore keep the same contract; this PR only adds the separate SuggestedAction column.

luoluoyuyu commented on iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/agent/task/PipeTaskAgent.java:
```diff
@@ -174,6 +172,13 @@ protected TPushPipeMetaRespExceptionMessage handleSinglePipeMetaChangesInternal(
+        .getRuntimeMeta()
+        .getConsensusGroupId2TaskMetaMap()
+        .values()
+        .forEach(PipeTaskMeta::clearExceptionMessages);
```
Clearing exceptions when applying meta from the coordinator prevents old CN-reported errors from overwriting the true last error on the DataNode after restart. Matches the do-not-auto-clear behavior for operators.

Caideyipi replied:
Agreed. This behavior is intentionally preserved: coordinator-provided exceptions are cleared before applying the metadata so they cannot overwrite the DataNode's locally tracked latest exception.

Caideyipi committed:
Merge master into fix-e

Caideyipi committed:
Fix pipe exception column compatibility and tests

Caideyipi commented:
Addressed the review and current deterministic CI failures in 0811b05ec85:

- preserved the existing Thrift exceptionMessage field and SHOW PIPES / information_schema ExceptionMessage / exception_message column names for client compatibility;
- added only the new SuggestedAction / suggested_action column;
- updated IoTDBDatabaseIT for the information-schema column list;
- updated IoTDBPipePermissionIT for the shifted remaining-event-count column.

The related pipe/meta unit tests pass. A focused local IT build compiled all relevant modules, but the locally injected com.timecho.ao plugin prevents JDBC startup by calling the removed IoTDBConfig.isEnableWhiteList() method, so the IT itself could not execute in this environment. The latest GitHub dual-table-manual-basic run is green.
